### PR TITLE
Limb grower

### PR DESCRIPTION
**Pull request by hamzah-hayat**
:cl: Limb Grower
wip: Added a limb grower machine to the game, uses synthflesh to produce limbs for humans/lizards, currently has sprites and basic code working, but due to the way limbs are initialised all limbs produced are currently invisible (though they still work and can be attached correctly)
/:cl:

I'm making this pull request because I'm having trouble finishing this due to the way body parts work. I've finished spriting and nearly all other code however.
Heres a look at the sprite:
limb-replicator

Its also animated, and has animations for filling up when creating limbs
TO DO
- make menus a little better (menu to purge/check chemicals)
- fix limbs appearing invisible
- adding more limbs/uses to the machine (perhaps even making the machine emagable)

### CODE REVIEW

**Comment by chickenboy10** [comment]
Oh bugger, i've messed up really badly

**Comment by optimumtact** [comment]
what are you concerned about?

**Comment by chickenboy10** [comment]
It says -126,238 lines in the corner, I guess i haven't merged stuff properly?

**Comment by optimumtact** [comment]
Did you touch the map?

if so you have to use the map merger https://tgstation13.org/wiki/Map_Merger as our maps are in a nonstandard format that reduces line conflicts

**Comment by AnturK** [comment]
Good idea.

**Comment by PKPenguin321** [comment]
sounds pretty neat but i wouldnt be shocked if it was denied for the same reason https://github.com/tgstation/tgstation/pull/20055 was denied

maybe make it really inefficient but allow science to upgrade it or something
also post pics of it on the map

File: code/game/machinery/limbgrower.dm [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+							"Lizard",
+							)
+
+	var/mob/living/carbon/human/human
+
+
+
+/obj/machinery/limbgrower/New()
+	..()
+	create_reagents(0)
```
Comment (AnturK) [inline code/game/machinery/limbgrower.dm]: You sure this is what you want ?

File: code/game/machinery/limbgrower.dm [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+/obj/item/weapon/circuitboard/machine/limbgrower
+	name = "circuit board (Limb Grower)"
+	build_path = /obj/machinery/limbgrower
+	origin_tech = "engineering=1;programming=1"
+	req_components = list(
+							/obj/item/weapon/stock_parts/manipulator = 1,
+							/obj/item/weapon/reagent_containers/glass/beaker = 2,
+							/obj/item/weapon/stock_parts/console_screen = 1)
+
+/obj/machinery/limbgrower/Destroy()
```
Comment (AnturK) [inline code/game/machinery/limbgrower.dm]: No need for this if you're not changing it.

**Comment by Shapsy** [comment]
As long as it takes a bit of setup and it would be more expedient to spend a few seconds looking for your arm that blew off before going to surgery, I don't think this would cause a large degree of power creep, this is a super cool idea

**Comment by InsaneHyena** [comment]
Reee, stop taking away robotics job

File: code/game/machinery/limbgrower.dm (hunk 1 above)
Comment (chickenboy10) [inline code/game/machinery/limbgrower.dm]: I think so?
I'm not quite sure how the reagents system works, so i put that in there (as the same is done on the circuit imprinter)

File: code/game/machinery/limbgrower.dm (hunk 2 above)
Comment (chickenboy10) [inline code/game/machinery/limbgrower.dm]: True

**Comment by chickenboy10** [comment]
I made this as i've watched many rounds where the roboticist was overwhelmed with a huge number of people coming in with missing limbs due to another double esword meme machine going around and chopping as many limbs of they could.
I want to keep this balanced by requiring synthflesh (a mid level chemical) from chemistry to produce.
Hopefully this is a long term solution to this, as the pull #20055 was literally just adding free limbs to medbay, while this is actually using chemistry to produce the limbs

**Comment by InsaneHyena** [comment]
There's already a limb producer in medbay, tbqh family, it's called genetics. You can even get a limb of your exact skin color, if you like. The only thing that genetics needs are monkey cubes.

**Comment by Kromgar** [comment]
You can regrow your limbs by turning into a monkey and back... that really should be fixed

**Comment by Bawhoppen** [comment]
don't make it upgradeable, but it needs to require more than synthflesh, make it require synflesh, blood, water, and if we had calcium as a chem I'd say that too but we don't

I'd also say slap one on the white ship aswell as thats supposed to be kind of a medical ship and could be interesting dynamic of fucking off into space to heal their arms after they got shot off by security

**Comment by Iamgoofball** [comment]
@Kromgar no way to fix it without it being completely shitcode to the max

File: code/game/machinery/limbgrower.dm [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+								L = new B()
+								world << "Trying to update limb"
+								L.name = "Synthentic Human Limb"
+								L.desc = "A synthentic human limb that will morph on its first use in surgery"
+							else if("Lizard")
+								//Lizard dna
+								L = new B()
+								world << "Trying to update lizard limb"
+								L.name = "Synthentic Lizard Limb"
+								L.desc = "A synthentic lizard limb that will morph on its first use in surgery"
```
Comment (RemieRichards) [inline code/game/machinery/limbgrower.dm]: No it won't, you can't use invalid bodyparts in invalid locations
arm in leg slot = failure.

**Comment by FantasticFwoosh** [comment]
Literally a few steps away from baymed bio-mass cloning/artificial organ creation.

File: code/game/machinery/limbgrower.dm [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+	src.updateUsrDialog()
+
+	return
+
+/obj/machinery/limbgrower/proc/build_limb(var/B)
+	//i need to create a body part manually using a set specias dna
+	var/obj/item/bodypart/L
+	var/selected_part
+	world << "category start"
```
Comment (optimumtact) [inline code/game/machinery/limbgrower.dm]: debug code left in by mistake?

File: code/game/machinery/limbgrower.dm [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+	L.name = "Synthentic [selected_category] Limb"
+	L.desc = "A synthentic [selected_category] limb that will morph on its first use in surgery. This one is for the [selected_part]"
+	L.loc = loc;
+
+/obj/machinery/limbgrower/RefreshParts()
+	reagents.maximum_volume = 0
+	for(var/obj/item/weapon/reagent_containers/glass/G in component_parts)
+		reagents.maximum_volume += G.volume
+		G.reagents.trans_to(src, G.reagents.total_volume)
+	var/T=1.2
```
Comment (optimumtact) [inline code/game/machinery/limbgrower.dm]: can you add a comment to indicate why the value is 1.2 ?

**Comment by optimumtact** [comment]
Stale, ask us to reopen this if you intend to continue working on it, or make a new pr from the same branch at this link https://github.com/tgstation/tgstation/compare/master...chickenboy10:Limb-Grower

**Comment by chickenboy10** [comment]
Thats fair enough, ive been working on it between other stuff, but it keeps having problems here and there.
Im also not fully set on what i want to do with it gameplay/balance wise.
ill reopen when im finished.

**Pull request closed without merging**